code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,783 @@
+
+	for(var/i in 1 to 2)
+		var/path = pick_n_take(medical)
+		new path(rune_turf)
+	. = ..()
+
+/obj/item/slimecross/warping/blue
+	colour = "blue"
+	runepath = /obj/effect/warped_rune/cyanspace //we'll call the blue rune cyanspace to not mix it up with actual bluespace rune
+	effect_desc = "Draw a rune that is slippery like water and may be activated to cover all adjacent tiles in ice."
+
+/obj/effect/warped_rune/cyanspace
+	icon_state = "rune_blue"
+	desc = "Its slippery like water and may be activated to cover all adjacent tiles in ice."
+
+/obj/effect/warped_rune/cyanspace/do_effect(mob/user)
+	for(var/turf/open/T in RANGE_TURFS(1, src) - rune_turf)
+		T.MakeSlippery(TURF_WET_PERMAFROST, 1 MINUTES)
+	. = ..()
+
+/obj/effect/warped_rune/cyanspace/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/slippery, 30)
+
+/obj/effect/warped_rune/cyanspace/Crossed(atom/movable/AM, oldloc)
+	if(isliving(AM))
+		activated = TRUE
+	. = ..()
+
+/obj/item/slimecross/warping/metal
+	colour = "metal"
+	runepath = /obj/effect/warped_rune/metalspace
+	effect_desc = "Draws a rune that may be activated to create a 3x3 block of invisible walls."
+
+//It's a wall what do you want from me
+/obj/effect/warped_rune/metalspace
+	desc = "This can be activated to to create a 3x3 block of invisible walls."
+	icon_state = "rune_metal"
+	density = TRUE
+
+/obj/effect/warped_rune/metalspace/do_effect(mob/user)
+	for(var/turf/open/T in RANGE_TURFS(1, src))
+		new /obj/effect/forcefield/mime(T)
+	. = ..()
+
+/obj/item/slimecross/warping/yellow
+	colour = "yellow"
+	runepath = /obj/effect/warped_rune/yellowspace
+	effect_desc = "Draw a rune that causes electrical interference."
+
+/obj/effect/warped_rune/yellowspace
+	desc = "Be careful with taking power cells with you!"
+	icon_state = "rune_yellow"
+
+/obj/effect/warped_rune/yellowspace/Crossed(atom/movable/AM, oldloc)
+	var/obj/item/stock_parts/cell/C = AM.get_cell()
+	if(!C && isliving(AM))
+		var/mob/living/L = AM
+		for(var/obj/item/I in L.GetAllContents())
+			C = I.get_cell()
+			if(C && C.charge)
+				break
+	if(C && C.charge)
+		do_sparks(5,FALSE,C)
+		for(var/mob/living/L in rune_turf)
+			electrocute_mob(L, C, src)
+		C.use(C.charge)
+		activated = TRUE
+	. = ..()
+
+/obj/item/slimecross/warping/darkpurple
+	colour = "dark purple"
+	runepath = /obj/effect/warped_rune/darkpurplespace
+	effect_desc = "Draw a rune that can transmute plasma into any other material."
+
+/obj/effect/warped_rune/darkpurplespace
+	icon = 'icons/obj/slimecrossing.dmi'
+	icon_state = "rune_dark_purple"
+	desc = "To gain something you must sacrifice something else in return."
+
+/obj/effect/warped_rune/darkpurplespace/do_effect(mob/user)
+	if(locate(/obj/item/stack/sheet/mineral/plasma) in rune_turf)
+		var/amt = 0
+		for(var/obj/item/stack/sheet/mineral/plasma/P in rune_turf)
+			amt += P.amount
+			qdel(P)
+		var/path_material = pick(subtypesof(/obj/item/stack/sheet/mineral) - /obj/item/stack/sheet/mineral/plasma)
```

Comment:
This is going to create any type of mineral sheets, e.g.

1. `/obj/item/stack/sheet/mineral/plasma/fifty` since you only removed `/obj/item/stack/sheet/mineral/plasma`; the amount will be overridden, though.
2. weird stuff like `/obj/item/stack/sheet/mineral/abductor`
3. slightly dangerous stuff like `/obj/item/stack/sheet/mineral/bananium`
4. unused stuff like `/obj/item/stack/sheet/mineral/mythril` (although it has a bit of meme potential)

You might want to build that list differently if you want to prevent plasma sheets from spawning; e.g. `subtypesof(/obj/item/stack/sheet/mineral) - typesof(/obj/item/stack/sheet/mineral/plasma)`